scipy/spatial/_spherical_voronoi.py

```diff
@@ -261,7 +262,7 @@ def sort_vertices_of_regions(self):
         """
         if self._dim != 3:
             raise TypeError("Only supported for three-dimensional point sets")
-        _voronoi.sort_vertices_of_regions(self._simplices, self.regions)
+        self.regions = sort_vertices_of_regions(self._simplices, self.regions)
```

Comment:
A few other things that might be helpful for me to understand:

1) How does the Pythran namespacing work with `sort_vertices_of_regions()` imported with the same name as the function it is defined in here?
2) Perhaps related--is there something about Pythran that makes it more useful to assign to `self.regions` than operating in place as before?
3) When Pythran is absent, how is the Cython `sort_vertices_of_regions` function going to be found with the `_voronoi` extension module qualifier removed from the call?

Reply:
You can safely ignore #1 because of the `self.` qualifier I think; #3 should go away if the imports shadow each other as noted above; #2 still seems like a reasonable query.

Reply:
Thanks for your comments! For `#2`, it is because I'm getting the warning when I compiled the file `WARNING: Exporting function 'sort_vertices_of_regions' that modifies its List argument. Beware that this argument won't be modified at Python call site`.

And Pythran doc says:
>In most cases (with the notable exception of numpy.ndarray), Pythran is working on a deep copy of the original Python arguments. This copy shares no memory relationship with the original object, which means that modifying the argument content in Pythran won’t modify the original argument content.

 So I  assign to `self.regions` to fix that, is that right?  @serge-sans-paille

Reply:
How are you going to do a drop in replacement for the Cython version of the function that doesn't return anything though, without modifying the Cython code (which would not be desirable)? 

Won't it return `None` to `self.regions` when Pythran isn't used?

Reply:
It probably isn't *that* big of a deal to modify the Cython code to return `regions` on top of modifying it in place--presumably that's a pass-through operation that doesn't introduce a copy. Bit awkward from a design standpoint maybe. Before modifying the Cython version (extra work) we'd probably want to confirm the performance gain.

Alternatively, maybe Pythran provides a runtime constant or attribute/hook for making a decision.

Reply:
That's indeed a problem. How about
```
tmp = _voronoi.sort_vertices_of_regions(self._simplices, self.regions)
if tmp != None: # use Pythran
    self.regions = tmp
```
That would make the code a little ugly though.

Reply:
Hmm, something like that maybe. I'll be less worried about ugliness if we're still getting a decent speedup when the dust settles, since it will be a path to getting rid of the Cython version at some point anyway, or at least pressuring the Cython version to get faster if it wants to stay around.